ast/src/parsed/asm.rs

```diff
@@ -37,19 +43,31 @@ impl Params {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct FunctionId<T> {
+    pub id: T,
+}
+
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub enum MachineStatement<T> {
     Degree(usize, AbstractNumberType),
     Submachine(usize, String, String),
     RegisterDeclaration(usize, String, Option<RegisterFlag>),
     InstructionDeclaration(usize, String, Params, InstructionBody<T>),
     InlinePil(usize, Vec<PilStatement<T>>),
-    FunctionDeclaration(usize, String, Params, Vec<FunctionStatement<T>>),
+    FunctionDeclaration(
+        usize,
+        String,
+        Option<FunctionId<T>>,
+        Params,
+        Vec<FunctionStatement<T>>,
+    ),
```

Comment:
Maybe leave a comment saying what these are? Unclear to me that the first argument is